src/ucp/api/ucp.h

```diff
@@ -644,6 +681,41 @@ typedef struct ucp_worker_params {
 
 
 /**
+ * @ingroup UCP_WORKER
+ * @brief Parameters for a UCP listener object.
+ *
+ * This structure defines parameters for @ref ucp_worker_listen, which is used to
+ * listen for incoming client/server connections.
+ */
+typedef struct ucp_worker_listener_params {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref ucp_worker_listener_params_field.
+     * Fields not specified in this mask would be ignored.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t                       field_mask;
+
+    /**
+     * An address in the form of a sockaddr.
+     * This field is mandatory for filling (along with its corresponding bit
+     * in the field_mask - @ref UCP_WORKER_LISTENER_PARAM_FIELD_SOCK_ADDR
+     * The ucp_worker_listen routine will return with an error if sock_address
+     * is not specified.
+     */
+    ucs_sock_addr_t                sock_address;
+
+    /**
+     * Handler to endpoint creation in a client-server connection flow.
+     * In order for the callback inside this handler to be invoked, the
+     * UCP_WORKER_LISTENER_PARAM_FIELD_CALLBACK needs to be set in the
+     * field_mask.
```

Comment:
What at the requirements for ucp_listener_accept_handler_t callback.
Does it have to be thread safe or signal safe. Can it call any other UCP function ?